pkg/daemon/update.go

```diff
@@ -1988,6 +1990,77 @@ func (dn *Daemon) cancelSIGTERM() {
 	}
 }
 
+func (dn *Daemon) SetLabelsOnNode(labels map[string]interface{}) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"labels": labels,
+		},
+	}
+	dn.logSystem("Setting labels %v on node %s", labels, dn.name)
+	patchData, err = json.Marshal(&patch)
+	if err != nil {
+		dn.logSystem("Error in setting labels on node %s: %v", dn.name, err)
+		return err
+	}
+	_, err = dn.kubeClient.CoreV1().Nodes().Patch(
+		context.TODO(), dn.name, types.MergePatchType, patchData, metav1.PatchOptions{})
+	if err != nil {
+		dn.logSystem("Error setting labels on node %s: %v", dn.name, err)
+		return err
+	}
+	return nil
+}
+
+func (dn *Daemon) setExcludeFromLBLabel(addLabel bool) error {
+	var labelValue interface{}
+	var err error
+	if addLabel {
```

Comment:
Can you explain what's happening here with this bool? having var called addLabel true/false makes it sound like a label won't be added if false, but it's only setting a label value? T Which itself seems confusing here bc there doesn't seem to be any way for labelValue to have a value..

Reply:
Tp just be clear, labelValue is declared above and is empty, if addLabel is true, it is..empty?

Reply:
code changed since last push, trying to resolve conversation, but github keeps giving me an error